### PR TITLE
[SPARK-32489][CORE] Pass `core` module UTs in Scala 2.13

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

So far, we fixed many stuffs in `core` module. This PR fixes the remaining UT failures in Scala 2.13.

- `OneApplicationResource.environmentInfo` will return a deterministic result for `sparkProperties`, `hadoopProperties`, `systemProperties`, and `classpathEntries`.
- `SubmitRestProtocolSuite` has Scala 2.13 answer in addition to the existing Scala 2.12 answer, and uses the expected answer based on the Scala runtime version. 

### Why are the changes needed?

To support Scala 2.13.

### Does this PR introduce _any_ user-facing change?

Yes, `environmentInfo` is changed, but this fixes the indeterministic behavior.

### How was this patch tested?

- Scala 2.12: Pass the Jenkins or GitHub Action
- Scala 2.13: Do the following.
```
$ dev/change-scala-version.sh 2.13
$ build/mvn test -pl core --am -Pscala-2.13
```

**BEFORE**
```
Tests: succeeded 2612, failed 3, canceled 1, ignored 8, pending 0
*** 3 TESTS FAILED ***
```

**AFTER**
```
Tests: succeeded 2615, failed 0, canceled 1, ignored 8, pending 0
All tests passed.
```

### CODE REVIEW

File: core/src/test/resources/HistoryServerExpectations/app_environment_expectation.json [hunk 1]
```diff
@@ -5,283 +5,283 @@
     "scalaVersion" : "version 2.11.8"
   },
   "sparkProperties" : [
```
Comment (dongjoon-hyun) [inline core/src/test/resources/HistoryServerExpectations/app_environment_expectation.json:7]: The change of this file is a simple sorting.

**Comment by dongjoon-hyun** [comment]
cc @srowen and @HyukjinKwon 
This is the last piece for `build/mvn test -pl core --am -Pscala-2.13`. After this PR, I'll add the test coverage into `GitHub Action` for this.

**Comment by HyukjinKwon** [comment]
Wow this is nice

File: core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala [hunk 2]
```diff
@@ -258,6 +260,33 @@ class SubmitRestProtocolSuite extends SparkFunSuite {
...
+      |  "clientSparkVersion" : "1.2.3",
+      |  "environmentVariables" : {
+      |    "PATH" : "/dev/null"
+      |  },
+      |  "mainClass" : "org.apache.spark.examples.SparkPie",
+      |  "sparkProperties" : {
+      |    "spark.driver.extraLibraryPath" : "pickle.jar",
+      |    "spark.jars" : "mayonnaise.jar,ketchup.jar",
+      |    "spark.driver.supervise" : "false",
+      |    "spark.driver.memory" : "1024m",
```
Comment (HyukjinKwon) [inline core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala:278]: Maybe `"${Utils.DEFAULT_DRIVER_MEM_MB}m"`?

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon . BTW, GitHub Action seems to be out of order for some reasons in both master branch and PR builder.

File: core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala (hunk 2 above)
Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala:278]: This is intentionally copied from `2.12` test case. If we change `2.13` here, we need to change `2.12`, too.
- https://github.com/apache/spark/pull/29298/files#diff-2ebc20516434bc6dc6ad4e9698e9b15bR239-R261

**Comment by HyukjinKwon** [comment]
Will take a look.

File: core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala (hunk 2 above)
Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala:278]: I'd like to avoid touching the existing Scala 2.12 test case~

**Comment by dongjoon-hyun** [comment]
Thanks. Take your time. This takes some time for review because it needs to run `core` and the dependencies.

File: core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala (hunk 2 above)
Comment (HyukjinKwon) [reply core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala:278]: Eh, https://github.com/apache/spark/pull/29298/files#diff-2ebc20516434bc6dc6ad4e9698e9b15bR239-R261 has `"${Utils.DEFAULT_DRIVER_MEM_MB}m"` here

Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala:278]: Oh, something looks incorrect. Let me check again.

Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala:278]: Got it. My bad.

Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala]: Thanks! It's fixed!

**Review by srowen — APPROVED** [review]
I was wondering if the order would matter to any user. It shouldn't. That said, if it does, the ordering isn't stable or guaranteed at the moment anyway. So seems OK to me.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
Thanks, @srowen . Yes. It does.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
Rebased to the master to bring @HyukjinKwon 's GitHub Action fixes.

**Comment by dongjoon-hyun** [comment]
All test passed except the `ExecutorSuite` which is known to be flaky in these day. I'll merge this. Thanks!

**Pull request closed without merging**